### PR TITLE
feat: Scalar ElysiaJS integration

### PR DESCRIPTION
**Pull request by marclave**
👋 hey everyone, i'm really excited for this PR and everyones feedback!

I think it would be pretty cool to expand the options of the swagger integration, so people can choose between Scalar & Swagger UI.

[Video Walkthrough](https://www.youtube.com/watch?v=Thuiz8D0g5w)
elysia (1)

Some big benefits of Scalar:
- Full customization
- an amazing ElysiaJS theme
- Integrated REST API Client for testing 
- Offline search using FUSE.js
- and many more small features

We would be beyond honored to have Scalar as the default, and I think i can create the integration in a way that its seamless and backwards compatible.

Also, what a joy it was working in ElysiaJS, the codebase is really clean and getting started was as simple as `bun dev` 🤯

Stoked to hear what y'all think.

I still need to:

- [x] add full configuration of scalar object
- [x] tidy up the theme 
- [x] add to the documentation
- [ ] iterate based on everyones feedback

### CODE REVIEW

File: src/types.ts
Comment (fecony) [inline src/types.ts:23]: jsdoc for `scalarVersion` and `provider` are swapped I believe

Comment (marclave) [reply src/types.ts:23]: great catch! just fixed :)

**Review by amjed-ali-k — APPROVED** [review]

**Review by SaltyAom — APPROVED** [review]
Sorry for the late reply, I have just got back from Christmas and New Year vacation.
First, let me address my gratitude.

Thank you and the Scalar team for the PR.
I know you and team take your time to make it great, and it means a lot to us.
So thank you a lot for your time.

It's very lovely, and I like it a lot.
We have been looking for an alternative to Swagger for a while and seeing one is refreshing.

For the PR, I think the PR is solid and will provide a necessary solid base for us.
And I think we can handle it from here.

There are many people excite about this PR and waiting to try it out, and to be honest, I'm also one of them.

**Therefore, I approve.**

If you want to apply any changes, feel free to do it, I will be waiting for your confirmation for merge.
Once the merge is complete, we will publish the package as soon as possible.

---
I have some feedback on my end but not related to the PR but about the Scalar (so we can get this merge first)

1. The request popup is very lovely and is there when I need it, a little problem is that if I input something and accidentally close and reopen it, the input will be reset.
Screenshot 2567-01-04 at 15 01 39

2 This is a suggestion, and not important, but I usually prefer fetch over axios but can't find it there, but that's alright.
Screenshot 2567-01-04 at 15 10 43

That's it, thank you for your time.

**Comment by marclave** [comment]
@SaltyAom thank you so much for your response, and time in writing out that message. I pushed up the conflict fixes, and this is ready to get merged :) I can help write out any documentation in the core ElysiaJS repository :) 

We are incredibly grateful for all the kind words, and how grateful we are for the community being excited for this PR. We are dedicated to making scalar the best, so we will always be here to fix and improve things fast!

Appreciate your feedback, i created issues for both of these https://github.com/scalar/scalar/issues/735 https://github.com/scalar/scalar/issues/737

I will look into these soon, incredibly excited to be integrated and to help out on the next steps once merged!

**Comment by hanspagel** [comment]
That feedback is so great! I’ve already added fetch support to our very own request example lib, we just need to add it to the API reference package. 😇 https://scalar-snippetz.netlify.app/

**Review by fecony — APPROVED** [review]

**Pull request merged**
